src/OpenSage.Game/Logic/Object/Body/BodyModule.cs

```diff
@@ -122,13 +121,13 @@ public virtual void ClearLastAttacker() { }
     public virtual bool FrontCrushed
     {
         get => false;
-        set => Debug.Fail("You should never call this for generic Bodys");
+        set => DebugUtility.Crash("You should never call this for generic Bodys");
```

Comment:
Is this reachable, or should this be abstract/have an abstract setter? Not really related to this PR, but it feels like a code smell (same with the below property)

Reply:
It is a bit strange. The getter is non-abstract, and returns the default value of `false`. The setter should really be abstract, but C#/.NET doesn't allow a non-abstract getter and an abstract setter.

Reply:
Yeah, by abstract setter I was more thinking an explicit abstract set method, since like you mentioned you can't have just an abstract setter on a property.